File: src/main/java/com/reroute/backend/locations/interfaces/LocationProvider.java
package com.reroute.backend.locations.interfaces;

import com.reroute.backend.model.distance.Distance;
import com.reroute.backend.model.location.DestinationLocation;
import com.reroute.backend.model.location.LocationPoint;
import com.reroute.backend.model.location.LocationType;

import java.util.List;

public interface LocationProvider {
    boolean isUsable();

    boolean isValidType(LocationType type);

    List<DestinationLocation> queryLocations(LocationPoint center, Distance range, LocationType type);

    void close();
}

File: src/main/scala/com/reroute/backend/locations/interfaces/LocationProvider.scala
package com.reroute.backend.locations.interfaces

import com.reroute.backend.locations.LocationsRequest
import com.reroute.backend.model.location.ReturnedLocation

trait LocationProvider {

  def queryLocations(requests: Seq[LocationsRequest]): Map[LocationsRequest, Seq[ReturnedLocation]]

  def validityFilter(req: LocationsRequest): Boolean

  def isUp: Boolean

  def close(): Unit
}
